src/corona_hakab_model/generation/matrix_generator.py

```diff
@@ -237,6 +239,15 @@ def _create_random_clustered_circles_matrix(self, con_type: ConnectionTypes, cir
             strengthes = np.random.choice(
                 [connection_strength, connection_strength / 7], size=len(conns), p=[daily_share, weekly_share]
             )
+            # check if some strengths were determined earlier
+            for conn in conns:
+                known_strength = known_strengths.get((conn, agent.index), None)
+                if known_strength != None:
+                    strengthes[np.where(conns==conn)] = known_strength
```

Comment:
wouldn't it be easier to enumerate conns and just address strengths by the index?
for i, conn in conns:
    strengths[i] = ...

anyway if this np.where works, i guess it doesnt really matters.